tensorflow/tensorflow.bzl

```diff
@@ -422,40 +429,79 @@ def tf_cc_shared_object(
         data = [],
         linkopts = [],
         framework_so = tf_binary_additional_srcs(),
+        soversion = None,
         kernels = [],
         per_os_targets = False,  # Generate targets with SHARED_LIBRARY_NAME_PATTERNS
         visibility = None,
         **kwargs):
+
+    if soversion != None:
+        suffix = "." + str(soversion).split(".")[0]
+        longsuffix = "." + str(soversion)
+    else:
+        suffix = ""
+        longsuffix = ""
+
     if per_os_targets:
-        names = [pattern % name for pattern in SHARED_LIBRARY_NAME_PATTERNS]
+        names = [
+            (pattern % (name, ''),
+                pattern % (name, suffix),
+                pattern % (name, longsuffix))
+            for pattern in SHARED_LIBRARY_NAME_PATTERNS
+        ]
     else:
-        names = [name]
-    for name_os in names:
+        names = [(
+            name,
+            name + suffix,
+            name + longsuffix
+        )]
+
+    for name_os, name_os_major, name_os_full in names:
+        if soversion != None:
+            native.genrule(
+                name = name_os + "_sym",
+                outs = [name_os],
+                srcs = [name_os_major],
+                output_to_bindir = 1,
+                cmd = "ln -sf $$(basename $<) $@",
```

Comment:
Symlink is not available on Windows by default, this won't work.

Reply:
done